interop-testing/src/main/java/io/grpc/testing/integration/TestServiceImpl.java

```diff
@@ -87,11 +88,33 @@ public void emptyCall(EmptyProtos.Empty empty,
    * Immediately responds with a payload of the type and size specified in the request.
    */
   @Override
-  public void unaryCall(SimpleRequest req,
-        StreamObserver<SimpleResponse> responseObserver) {
+  public void unaryCall(SimpleRequest req, StreamObserver<SimpleResponse> responseObserver) {
+    ServerCallStreamObserver<SimpleResponse> obs =
+        (ServerCallStreamObserver<SimpleResponse>) responseObserver;
     SimpleResponse.Builder responseBuilder = SimpleResponse.newBuilder();
+    switch (req.getResponseCompression()) {
+      case DEFLATE:
+        // fallthrough, just use gzip
+      case GZIP:
+        obs.setCompression("gzip");
+        obs.setMessageCompression(true);
+        break;
+      case NONE:
+        obs.setCompression("identity");
+        obs.setMessageCompression(false);
+        break;
+      case UNRECOGNIZED:
+        // fallthrough
+      default:
+        obs.onError(Status.UNIMPLEMENTED
+            .withDescription("Unknown: " + req.getResponseCompression())
+            .asRuntimeException());
+        return;
+    }
+
+    boolean compressable = false;
     if (req.getResponseSize() != 0) {
-      boolean compressable = compressableResponse(req.getResponseType());
```

Comment:
Oh, previously it was using compressible to decide whether or not to compress.  That is now gone with the new proto.  Fixed.